src/coreclr/minipal/Windows/dn-stdio.cpp

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <Windows.h>
+#include <dn-stdio.h>
+
+int fopen_u16(FILE** stream, const WCHAR* path, const WCHAR* mode)
+{
+    return _wfopen_s(stream, path, mode);
+}
+
+int64_t fgetsize(FILE* stream)
+{
+    fpos_t current;
+    fgetpos(stream, &current);
+    fseek(stream, 0, SEEK_END);
+    int64_t length = _ftelli64(stream);
+    fsetpos(stream, &current);
+    return length;
+}
+
+int64_t ftell_64(FILE* stream)
+{
+    fpos_t pos;
+    fgetpos(stream, &pos);
+    return (int64_t)pos;
+}
+
+int fsetpos_64(FILE* stream, int64_t pos)
+{
+    fpos_t fpos = (fpos_t)pos;
+    return fsetpos(stream, &fpos);
+}
+
+HRESULT HRESULT_FROM_LAST_STDIO()
+{
+    return HRESULT_FROM_WIN32(::GetLastError());
```

Comment:
Is GetLastError correct here? C runtime methods are documented to return error in errno.

Reply:
> CRT functions should also be preserving the LastError when calling Windows API

They happen to not stomp on the last error most of the time. I do not think they are preserving it intentionally 100%.

The errno mapping and the long path support parts of this PR are not pretty. I have not figured out what to do about it yet...